ersilia/core/tracking.py

```diff
@@ -368,11 +392,17 @@ def track(self, input, result, meta):
         time = datetime.now() - self.time_start
         json_dict["time_taken"] = str(time)
 
+         # Convert result_dataframe (list of dicts) to columns (dict of lists)
+        columns = defaultdict(list)
+        for row in result_dataframe:
+            for key, value in row.items():
+                columns[key].append(value)
```

Comment:
I think you can do this conversion to defaultdict within the check_types method because that's where it's really needed. The `track` method does not need to know about this conversion.